mesmer/core/localized_covariance.py

```diff
@@ -0,0 +1,227 @@
+import numpy as np
+from scipy.stats import multivariate_normal
+
+
+def _adjust_ecov_ar1_np(covariance, ar_coefs):
+    """
+    adjust localized empirical covariance matrix for autoregressive process of order 1
+
+    Parameters
+    ----------
+    covariance : 2D np.array
+        Empirical covariance matrix.
+    ar_coefs : 1D np.array
+        The coefficients of the autoregressive process of order 1.
+
+    Returns
+    -------
+    adjusted_covariance : np.array
+        Adjusted empirical covariance matrix.
+
+    Notes
+    -----
+    - Adjusts ``covariance`` for an AR(1) process according to [1]_, eq (8).
+
+    - The formula is specific for an AR(1) process, see also `#167 (comment)
+      https://github.com/MESMER-group/mesmer/pull/167#discussion_r912481495`__.
+
+    - According to [2]_ "The multiplication with the ``reduction_factor`` scales the
+      empirical standard error under the assumption of an autoregressive process of
+      order 1 [3]_. This accounts for the fact that the variance of an autoregressive
+      process is larger than that of the driving white noise process."
+
+    - This formula is wrong in [1]_. However, it is correct in the code. See also [2]_
+       and [3]_.
+
+    .. [1] Beusch, L., Gudmundsson, L., and Seneviratne, S. I.: Emulating Earth system model
+       temperatures with MESMER: from global mean temperature trajectories to grid-point-
+       level realizations on land, Earth Syst. Dynam., 11, 139–159,
+       https://doi.org/10.5194/esd-11-139-2020, 2020.
+
+    .. [2] Humphrey, V. and Gudmundsson, L.: GRACE-REC: a reconstruction of climate-driven
+       water storage changes over the last century, Earth Syst. Sci. Data, 11, 1153–1170,
+       https://doi.org/10.5194/essd-11-1153-2019, 2019.
+
+    .. [3] Cressie, N. and Wikle, C. K.: Statistics for spatio-temporal data, John Wiley
+       & Sons, Hoboken, New Jersey, USA, 2011.
+    """
+
+    reduction_factor = np.sqrt(1 - ar_coefs**2)
+    reduction_factor = np.atleast_2d(reduction_factor)  # so it can be transposed
+
+    # equivalent to ``diag(reduction_factor) @ ecov @ diag(reduction_factor)``
+    return reduction_factor * reduction_factor.T * covariance
```

Comment:
```suggestion
    return diag(reduction_factor) @ covariance @ diag(reduction_factor)
```

Is there some reason we don't do this?

Reply:
I find it easier to read this way

Reply:
Fair enough. I guess I find the other way easier because I don't have to think, "Does changing the order of multiplication of these matrices matter?" We've got tests though so doesn't matter much either way.